x-pack/plugins/security_solution/public/management/components/hooks/use_is_mounted.ts

```diff
@@ -5,22 +5,22 @@
  * 2.0.
  */
 
-import { useEffect, useRef } from 'react';
+import { useEffect, useState } from 'react';
 
 /**
- * Track when a comonent is mounted/unmounted. Good for use in async processing that may update
+ * Track when a component is mounted/unmounted. Good for use in async processing that may update
  * a component's internal state.
  */
 export const useIsMounted = (): boolean => {
-  const isMounted = useRef(false);
+  const [isMounted, setIsMounted] = useState(false);
```

Comment:
Why did you changed this from useRef to useState?

Reply:
good question. This was a bug in this hook.
because `useRef()` returns a mutable object, when we update back to `false`, it actually does nothing to notify consumers of it to indicate that the component is unmounted. thats because `isMounted.current = false` does not actually trigger a re-render.

So changing that to `useState()` will cause a re-render because its managed component state.

Reply:
Awesome, thanks!